tests/test_communication_manager.py

```diff
@@ -110,19 +121,31 @@ def test_communication_manager_can_initialize_and_train_and_average_model():
         RawEventTypes.NEW_INFO.name,
         new_weights_event
     )
-    while len(scheduler.processed) == 3:
+    timeout = time.time() + 3
+    while time.time() < timeout and len(scheduler.processed) == 3:
+        # averaging job
         scheduler.runners_run_next_jobs()
         time.sleep(0.1)
+    # we've only heard one set of new weights so our listen_iters are 1
     assert communication_manager.optimizer.listen_iterations == 1
```

Comment:
Same as above, can't tell whether assert statement fails because of timeout or runner error.

Reply:
Done